tests/PayPal/Test/Auth/OAuthTokenCredentialTest.php

```diff
@@ -60,6 +60,27 @@ public function testGetAccessTokenUnit()
         $this->assertNotNull($result);
     }
 
+    public function testGetAccessTokenWithSubjectUnit()
+    {
+        $config = array(
+            'mode' => 'sandbox',
+            'cache.enabled' => true,
+            'cache.FileName' => AuthorizationCacheTest::CACHE_FILE
+        );
+        $cred = new OAuthTokenCredential('clientId', 'clientSecret', 'subject');
+
+        //{"clientId":{"clientId":"clientId","accessToken":"accessToken","tokenCreateTime":1421204091,"tokenExpiresIn":288000000}}
+        AuthorizationCache::push($config, 'clientId', $cred->encrypt('accessTokenWithSubject'), 1421204091, 288000000);
```

Comment:
If `subject` is relevant only for third party tokens (refresh tokens), we should not cache the response. This could lead to all consequent calls with just a `clientId` to use this `subject` based access token. This will be because, we only fetch this cache by `clientId` and nothing else.